packages/blockchain-wallet-v4-frontend/src/data/session/slice.ts

```diff
@@ -0,0 +1,25 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { dissoc, mergeRight } from 'ramda'
+
+import { SessionStateType } from './types'
+
+const initialState: SessionStateType = {}
+
+const sessionSlice = createSlice({
+  initialState,
+  name: 'session',
+  reducers: {
+    removeSession: (state, action: PayloadAction<string>) => {
+      state = dissoc(action.payload, state.sessions)
```

Comment:
This is totally fine, but you also don't have return a new object here if you don't want. You can modify the state object like `delete state[action.payload]` but understandable if you don't want to do it that way since `delete` has it's own problems.